modules/extensions-spring-support/src/main/java/org/mule/module/extension/internal/config/XmlExtensionParserUtils.java

```diff
@@ -84,6 +93,183 @@ static void parseConfigName(Element element, BeanDefinitionBuilder builder)
+    }
+
+    /**
+     * Parses an {@code element} which is assumed to be a representation of the given
+     * {@code dataType}. It then returns a {@link ValueResolver} which will provide
+     * the actual value extracted from the {@code element}
+     *
+     * @param element      a {@link ElementDescriptor}
+     * @param fieldName    the name of the variable in which the value is to be stored
+     * @param dataType     the {@link DataType} that describes the type of the value to be extracted
+     * @param defaultValue a default value in case that the {@code element} does not contain an actual value
+     * @return a {@link ValueResolver}
+     */
+    static ValueResolver parseElement(final ElementDescriptor element,
+                                      final String fieldName,
+                                      final DataType dataType,
+                                      final Object defaultValue)
+    {
+        final String hyphenizedFieldName = hyphenize(fieldName);
+        final String singularName = singularize(hyphenizedFieldName);
+        final ValueHolder<ValueResolver> resolverReference = new ValueHolder<>();
+
+        DataQualifierVisitor visitor = new AbstractDataQualifierVisitor()
+        {
+
+            /**
+             * An attribute of a generic type
+             */
+            @Override
+            public void defaultOperation()
+            {
+                resolverReference.set(getResolverFromValue(getAttributeValue(element, fieldName, defaultValue), dataType));
+            }
+
+            /**
+             * A collection type. Might be defined in an inner element or referenced
+             * from an attribute
+             */
+            @Override
+            public void onList()
+            {
+                resolverReference.set(parseCollection(element, fieldName, hyphenizedFieldName, singularName, defaultValue, dataType));
+            }
+
+            @Override
+            public void onPojo()
+            {
+                resolverReference.set(parsePojo(element, fieldName, hyphenizedFieldName, dataType, defaultValue));
+            }
+
+            @Override
+            public void onDateTime()
+            {
+                if (Calendar.class.isAssignableFrom(dataType.getRawType()))
+                {
+                    resolverReference.set(parseCalendar(element, fieldName, dataType, defaultValue));
+                }
+                else
+                {
+                    resolverReference.set(parseDate(element, fieldName, dataType, defaultValue));
+                }
+            }
+        };
+
+        dataType.getQualifier().accept(visitor);
+        return resolverReference.get();
+    }
+
+    /**
+     * Creates a {@link BeanDefinition} which generates a {@link ElementDescriptor}
+     * that describes the given {@code element}
+     *
+     * @param element an {@link Element}
```

Comment:
actually I could not... This is pretty clear... an Element goes in, an ElementDescritor comes out